discovery-provider/src/utils/redis_cache.py

```diff
@@ -27,11 +26,11 @@ def use_redis_cache(key, ttl_sec, work_func):
 
     if cached_value:
         logger.info(f"Redis Cache - hit {key}")
-        return json.loads(cached_value)
+        return pickle.loads(cached_value)
```

Comment:
maybe put a try catch here too, cause it's going to break when we switch it over from json.

Reply:
ty